advanced/src/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { HttpLink, InMemoryCache, ApolloClient } from 'apollo-client-preset'
+import { WebSocketLink } from 'apollo-link-ws'
+import { ApolloLink, split } from 'apollo-link'
+import { getMainDefinition } from 'apollo-utilities'
+import { AUTH_TOKEN } from './constant'
+import SuperContainer from './components/SuperContainer'
+
+import 'tachyons'
+import './index.css'
+
+const httpLink = new HttpLink({ uri: 'http://localhost:4000' })
+
+const middlewareLink = new ApolloLink((operation, forward) => {
+  // get the authentication token from local storage if it exists
+  const tokenValue = localStorage.getItem(AUTH_TOKEN)
+  // return the headers to the context so httpLink can read them
+  operation.setContext({
+    headers: {
+      Authorization: tokenValue ? `Bearer ${tokenValue}` : "",
+    }
+  })
+  return forward(operation)
+})
+
+// Authenticated httplink
+const httpLinkAuth = middlewareLink.concat(httpLink)
+
+const wsLink = new WebSocketLink({
+  uri: `wss://localhost:4000`,
+  options: {
+    reconnect: true,
+    connectionParams: {
+      Authorization: `Bearer ${localStorage.getItem(AUTH_TOKEN)}`
+    }
+  }
+})
+
+const link = split(
+  // split based on operation type
+  ({ query }) => {
+    const { kind, operation } = getMainDefinition(query)
+    return kind === 'OperationDefinition' && operation === 'subscription'
+  },
+  wsLink,
+  httpLinkAuth,
+)
+
+// apollo client setup
+const client = new ApolloClient({
+  link: ApolloLink.from([link]),
+  cache: new InMemoryCache(),
+  connectToDevTools: true
```

Comment:
Is this something new I don't know about @manjula91? 😄

Reply:
I have used apollo client 2.0

https://github.com/apollographql/apollo-client/blob/master/docs/source/2.0-migration.md

I think you can take a look at this docs

Reply:
Is this a problem on any side ?